src/snapred/backend/service/NormalizationService.py

```diff
@@ -162,31 +165,52 @@ def normalizationAssessment(self, request: NormalizationRequest):
             peakIntensityThreshold=request.peakIntensityThreshold,
             crystalDBounds=request.crystalDBounds,
         )
-        calibration = self.sousChef.prepCalibration(farmFresh)
+        normalization = parse_obj_as(Normalization, self.sousChef.prepCalibration(farmFresh))
         record = NormalizationRecord(
             runNumber=request.runNumber,
             useLiteMode=request.useLiteMode,
             peakIntensityThreshold=request.peakIntensityThreshold,
             backgroundRunNumber=request.backgroundRunNumber,
             smoothingParameter=request.smoothingParameter,
-            calibration=calibration,
+            calculationParameters=normalization,
             crystalDBounds=request.crystalDBounds,
         )
         return record
 
     @FromString
     def saveNormalization(self, request: NormalizationExportRequest):
+        """
+        If no version is attached to the request, this will save at next version number
+        """
         entry = request.normalizationIndexEntry
-        version = entry.version
-        normalizationRecord = request.normalizationRecord
-        if version is not None:
-            normalizationRecord.version = version
-        normalizationRecord = self.dataExportService.exportNormalizationRecord(normalizationRecord)
-        normalizationRecord = self.dataExportService.exportNormalizationWorkspaces(normalizationRecord)
-        entry.version = normalizationRecord.version
+        record = request.normalizationRecord
+        version = self.dataFactoryService.getThisOrNextNormalizationVersion(
```

Comment:
as per my calibration comment on explicit failure.

Reply:
I can add a `validate_call` tag to the method in `Indexer`, and that will cause an error if anything other than `int` or `None` is passed

Reply:
At this point we expect either:
1. version is None, or
2. version was set by the user.
If the version is None, this should set it to the next version.